ocaml/xapi/cli_operations.ml

```diff
@@ -2843,15 +2843,34 @@ let vm_migrate printer rpc session_id params =
              let gpu_group = Client.GPU_group.get_by_uuid remote_rpc remote_session gpu_group_uuid in
              vgpu,gpu_group) (read_map_params "vgpu" params) in
 
-         let default_sr =
-           try let pools = Client.Pool.get_all remote_rpc remote_session in
-             printer (Cli_printer.PMsg "Selecting remote pool's default SR for migrating VDIs") ;
-             Some (Client.Pool.get_default_SR remote_rpc remote_session (List.hd pools))
+         let preferred_sr =
+           try
+             let query = Printf.sprintf "(field \"host\"=\"%s\") and (field \"currently_attached\"=\"true\")" (Ref.string_of host) in
+             let host_pbds = Client.PBD.get_all_records_where remote_rpc remote_session query in
+             let srs = List.map (fun (pbd_ref, pbd_rec) -> sr_record remote_rpc remote_session pbd_rec.API.pBD_SR) host_pbds in
```

Comment:
The use of the `sr_record` function (from records.ml) is a little confusing here, and also does a lot more work than you actually need. Just use `Client.SR.get_record`.